src/store.js

```diff
@@ -139,24 +139,43 @@ export const useStore = defineStore("store", {
           this.isInputDisabled = true;
         });
     },
+
+    resetToDefault() {
+      this.lastBreedState = [];
+      this.indexClicked = 0;
+    },
+    resetSearchQuery() {
+      this.searchQuery = "";
+    },
     toggleCard(breed, id) {
+      this.setBreedIsActive(this.ifBreedActive(this.toggleIsActive(breed)));
+      this.addBreedState(id);
+    },
+
+    toggleIsActive(breed) {
       breed.isActive = !breed.isActive;
+      return breed;
+    },
+    ifBreedActive(breed) {
       if (breed.isActive) {
         this.indexClicked++;
-        this.lastBreedState.push(id);
+        return breed;
       }
+      return breed;
+    },
+    setBreedIsActive(breed) {
       this.displayedList.forEach((item) => {
         if (item.name !== breed.name) {
           item.isActive = false;
+          return item;
         }
+        return item;
       });
     },
-    resetToDefault() {
-      this.lastBreedState = [];
-      this.indexClicked = 0;
-    },
-    resetSearchQuery() {
-      this.searchQuery = "";
+    addBreedState(id) {
+      if (this.displayedList[id].isActive) {
+        this.lastBreedState.push(id);
+      }
     },

```

Comment:
Nekako mi ni ratalo, da bi imel samo en argument v function composition, zato, ker v tej `addBreedState `res rabim ta `id`, zato kličem potem v `toggleCard `metodi to fn posebej in ne znotraj function comositiona

```
toggleCard(breed, id) {
      this.setBreedIsActive(this.ifBreedActive(this.toggleIsActive(breed)));
      this.addBreedState(id);
    },
```

Reply:
ok, ni panike 👍